aptos-move/block-executor/src/executor.rs

```diff
@@ -1188,15 +1216,28 @@ where
                 executor_arguments,
                 signature_verified_block,
                 base_view,
+                true,
             )
         };
 
         if let Err(Error::FallbackToSequential(e)) = ret {
-            if let PanicOr::Or(ModulePathReadWrite) = e {
-                debug!("[Execution]: Module read & written, sequential fallback");
-            } else {
-                error!("[Execution]: {:?}, sequential fallback", e);
-            }
+            let dynamic_change_set_optimizations_enabled = match e {
+                PanicOr::Or(IntentionalFallbackToSequential::ModulePathReadWrite) => {
+                    debug!("[Execution]: Module read & written, sequential fallback");
+                    true
```

Comment:
What's the idea behind enabling the `dynamic_change_set_optimizations_enabled` flag when ModulePathReadWrite error and disabling the flag for the other two error types?

Reply:
ModulePathReadWrite happens often, and code should be able to handle it with dynamic_change_set_optimizations_enabled=true (and that we test/exercise more of the code)

But, I agree, I am gong to do this fallback for code_invariant_error as well. I am refactoring this code a bit to make it clear